articles/proofs-arguments-and-zero-knowledge/chapter_19.md

```diff
@@ -0,0 +1,140 @@
+---
+Title: Bird’s Eye View of Practical Arguments
+Status: In Progress
+Level: "6"
+---
+# Chapter 19: Bird’s Eye View of Practical Arguments
+
+![](attachments/taxonomy.png)
+
+## 19.1 A Taxonomy of SNARKs
+
+Outside of the **linear-PCP-based SNARKs**, most known **SNARKs** are obtained by combining some **IP**,  **MIP**, or **constant-round polynomial IOP** with a **polynomial commitment** scheme.
+
+For example:
+
+| Polynomial IOP approaches      | Polynomial Commitment Approaches                                                     | SNARK                                            |
+| ------------------------------ | ------------------------------------------------------------------------------------ | ------------------------------------------------ |
+| IPs                            | FRI-based (multilinear) polynomial commitments                                       | Virgo                                            |
+| IPs                            | Discrete-log-based (multilinear) polynomial commitments (Bulletproofs, Hyrax-commit) | Hyrax                                            |
+| IPs                            | KZG-based (multilinear) polynomial commitments                                       | zk-vSQL and Libra                                |
+| MIPs                           | Multilinear polynomial commitments                                                   | Spartan, Xiphos, Kopis, Brakedown, and Shockwave |
+| Constant-round polynomial IOPs | FRI-based (univariate) polynomial commitments                                        | Aurora, Fractal, and Redshift                    |
+| Constant-round polynomial IOPs | KZG-based (univariate) polynomial commitments                                        | Marlin                                           |
+| Constant-round polynomial IOP  | Ligero                                                                               | Ligero                                           |
+
+Beside that, the most popular variant of the SNARK derived from GGPR’s linear PCP is
+Groth16.
+
+**More SNARKs via composition**: On top of the taxonomy of SNARKs delineated above, one can take any two SNARKs designed via one of the above approaches, and compose them one or more times. Such compositions are growing increasingly popular and already yield state-of-the-art performance.
+
+## 19.2 Pros and Cons of the Approaches
+
+Sections 10.6 discussed the pros and cons of the **IP, MIP,** and **constant-round polynomial IOP**, while sections 16.3 discussed those of polynomial commitment scheme. In this section, our aim is to do the same for the various combinations and for SNARKs derived from linear PCPs.
+
+### Approaches Minimizing Proof Size
+
+There are 2 approaches that achieve proofs consisting of a constant
```

Comment:
Add something like this for clearer explanation:
proofs consisting of a constant number (i.e the smallest proof size) of group elements